include/pokemon.h

```diff
@@ -337,17 +337,27 @@ struct SpeciesInfo /*0x24*/
 struct BattleMove
 {
     u16 effect;
-    u16 power;  //higher than 255 for z moves
-    u8 type;
+    u8 power;  //higher than 255 for z moves
+    u8 type:5;
+    u8 recoil:3;
+
     u8 accuracy;
-    u8 pp;
     u8 secondaryEffectChance;
-    u16 target;
     s8 priority;
+    u8 multihit:4;
+    u8 critBoost:2;
+    u8 split:2;
+
+    u16 target;
+    u8 pp;
+    union {
+        u8 effect;
+        u8 powerOverride;
+    } zMove;
+
     u32 flags;
-    u8 split;
+
     u16 argument;
-    u8 zMoveEffect;
 };
```

Comment:
Preliminary thoughts:
- `recoil` doesn't have enough bits to fit all the values possible by `RECOIL_PERCENTAGE`, like 1%.
- Fields should use either defines or numbers, but not both like how `multihit` is doing.